src/components/input-text/input-text.tsx

```diff
@@ -0,0 +1,554 @@
+    origin: SetValueOrigin;
+    previousValue?: string;
+    value: string;
+  }): void => {
+    this.setPreviousValue(previousValue || this.value);
+    this.previousValueOrigin = origin;
+    this.userChangedValue = origin === "user";
+    this.value = value;
+
+    if (origin === "direct") {
+      this.setInputValue(value);
+    }
+
+    if (nativeEvent) {
+      const calciteInputTextInputEvent = this.calciteInputTextInput.emit({
+        element: this.childEl,
+        nativeEvent,
+        value: this.value
+      });
+
+      if (calciteInputTextInputEvent.defaultPrevented) {
+        this.value = this.previousValue;
+      } else if (committing) {
+        this.emitChangeIfUserModified();
+      }
+    }
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render(): VNode {
+    const dir = getElementDir(this.el);
+    const loader = (
+      <div class={CSS.loader}>
+        <calcite-progress label={this.intlLoading} type="indeterminate" />
+      </div>
+    );
+
+    const inputClearButton = (
+      <button
+        aria-label={this.intlClear || TEXT.clear}
+        class={CSS.clearButton}
+        disabled={this.disabled || this.readOnly}
+        onClick={this.clearInputValue}
+        tabIndex={this.disabled ? -1 : 0}
+        type="button"
+      >
+        <calcite-icon icon="x" scale="s" />
+      </button>
+    );
+    const iconEl = (
+      <calcite-icon
+        class={CSS.inputIcon}
+        flipRtl={this.iconFlipRtl}
+        icon={this.requestedIcon}
+        scale="s"
+      />
+    );
+    const prefixText = <div class={CSS.prefix}>{this.prefixText}</div>;
+
+    const suffixText = <div class={CSS.suffix}>{this.suffixText}</div>;
+
+    const childEl = (
+      <input
+        aria-label={getLabelText(this)}
+        autofocus={this.autofocus ? true : null}
+        class={{
+          [CSS.editingEnabled]: this.editingEnabled,
+          [CSS.inlineChild]: !!this.inlineEditableEl
+        }}
+        defaultValue={this.defaultValue}
+        disabled={this.disabled ? true : null}
+        enterKeyHint={this.el.enterKeyHint}
+        inputMode={this.el.inputMode}
+        maxLength={this.maxLength}
+        minLength={this.minLength}
+        name={this.name}
+        onBlur={this.inputBlurHandler}
+        onFocus={this.inputFocusHandler}
+        onInput={this.inputInputHandler}
+        onKeyDown={this.inputKeyDownHandler}
+        placeholder={this.placeholder || ""}
+        readOnly={this.readOnly}
+        ref={this.setChildElRef}
+        required={this.required ? true : null}
+        tabIndex={this.disabled || (this.inlineEditableEl && !this.editingEnabled) ? -1 : null}
+        type="text"
+        value={this.value}
+      />
+    );
+
+    return (
+      <Host onClick={this.inputFocusHandler} onKeyDown={this.keyDownHandler}>
```

Comment:
☝️ I'm also thinking these two handlers should be named `host${eventName}Handler`.